kgforge/specializations/models/rdf/rdf_model_service_from_directory.py

```diff
@@ -73,18 +72,31 @@ def _build_shapes_map(self) -> Dict:
                         ?shape a rdfs:Class
                     }
                 }
-            } ORDER BY ?type"""
+            } ORDER BY ?type
+        """
         res = self._graph.query(query)
-        return {row["type"]: row["shape"] for row in res}
 
+        class_being_shaped_id_to_shape_uri: Dict[str, URIRef] = {
+            row["type"]: URIRef(row["shape"])
```

Comment:
Why are propagating URIRef instead of str ?

Reply:
https://github.com/BlueBrain/nexus-forge/blob/master/kgforge/specializations/models/rdf/store_service.py#L103 it's a consequence of this line. The dictionary value was set to be a URIRef for class_to_shapes and this is used by _build_types_to_shapes to create types_to_shapes and get_shape_from_type gets it from it. So I just added types